pkgs/development/python-modules/jax/default.nix

```diff
@@ -49,16 +48,16 @@ buildPythonPackage rec {
   # jaxlib is _not_ included in propagatedBuildInputs because there are
   # different versions of jaxlib depending on the desired target hardware. The
   # JAX project ships separate wheels for CPU, GPU, and TPU.
-  propagatedBuildInputs = [
+  dependencies = [
+    jaxlib
```

Comment:
Is this comment and https://github.com/NixOS/nixpkgs/pull/156808 no longer relevant now that jax depends on jaxlib directly with the [CUDA plugin change](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-4-30-june-18-2024) in `0.4.30`? I was sitting on a PR removing jaxlib from {dm-haiku, equinox, flax, optax, etc.} but I guess it's no longer relevant. And should packages depending on `jax` also put `jaxlib` in their `dependencies` after this PR or is that no longer necessary as `jax` itself now depends on `jaxlib`?

Thanks for tackling this by the way!

Reply:
> Is this comment and https://github.com/NixOS/nixpkgs/pull/156808 no longer relevant now that jax depends on jaxlib directly with the [CUDA plugin change](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-4-30-june-18-2024) in 0.4.30?

That is also my understanding

> I was sitting on a PR removing jaxlib from {dm-haiku, equinox, flax, optax, etc.} but I guess it's no longer relevant. And should packages depending on jax also put jaxlib in their dependencies after this PR or is that no longer necessary as jax itself now depends on jaxlib?

I agree that this situation is a bit confusing currently. I propose that we merge this PR along with written guidance to maintainers on migrating.

Reply:
Yes, from my understanding, we will be able to get rid of specifying `jaxlib` in downstream packages.

Reply:
I'll delete this comment as it causes confusion.

> I propose that we merge this PR along with written guidance to maintainers on migrating.

Do you mean to add in the release notes that it is not necessary to explicitly add `jaxlib` as a dependency for packages that depend on `jax`?

Reply:
I added the information to the release note.
https://github.com/NixOS/nixpkgs/pull/363130/commits/b69c1a410ef9dde6999e3321a55133d776bec347